networkx/algorithms/bipartite/extendability.py

```diff
@@ -0,0 +1,158 @@
+""" Provides a function for computing the extendability of
+graph which is undirected, simple, connected and bipartite.
+
+For example:
+
+>> nodes = ['1', '2', '3', '4',
+            '5', '6', '7', '8']
+>> edges = [('1', '5'), ('1', '6'), ('1', '7'),
+            ('2', '5'), ('2', '6'), ('2', '8'),
+            ('3', '6'), ('3', '7'), ('3', '8'),
+            ('4', '5'), ('4', '7'), ('4', '8')]
+>> graph = nx.Graph()
+>> graph.add_nodes_from(nodes)
+>> graph.add_edges_from(edges)
+>> find_extendability(graph)
+2
+
+"""
+
+
+import networkx as nx
+from networkx.algorithms.components.connected import is_connected
+from networkx.algorithms.bipartite import is_bipartite, sets
+from networkx.algorithms.bipartite.matching import hopcroft_karp_matching
+from networkx.algorithms.matching import is_perfect_matching
+from networkx.algorithms.components import is_strongly_connected
+from networkx.algorithms.connectivity.disjoint_paths import node_disjoint_paths
```

Comment:
Instead of importing each function from nx that you use, use the single nx import and call these functions with e.g.  `nx.is_connected`.  The functions from the bipartitie subpacke need that prefix (becuase so many have the same name as the non-bipartitie version). So use `nx.bipartite.hopcraft_karp_matching` and `nx.bipartite.is_bipartite`.

Reply:
Since i am new to git, i would like to ask some questions. I will make these changes in my local computer, won't i? If yes, then do i have to push the changes into a new PR?

Reply:
Questions about git are fine! :}   
Just commit the new changes to the same branch locally and push that branch to your repository on github.  Github automatically pushes any changes to you branch on your github repository to the PR and the tests on github are run again. That also notifies anyone who is "following" your PR with an email that there are new things to look at.

Reply:
So, it is not necessary to press the "compare" button for creating a new PR.
If the above notation is correct, then i am ready to proceed.
Thank you !!!

Reply:
One last question:
Do i follow the standard procedure ? git add, git commit, git push
Is it required to write a message on "git commit" command ?

Reply:
Yes, 
`git commit -a` does the add for you if you aren't adding any new files (the -a means to add any *modified* files). So you can use:  
```
git commit -a
git push.
```

It is not required to add a message, but it is often helpful to keep track of which commits did which changes.  You can add the message on the command line using:  
```git commit -am "updated tests"```

Reply:
Unfortunately tests failed ! What is the problem ?

Reply:
When you click on "Details" to the right of the failed test, it brings you to a page that shows
the full test log -- starting at the end where the error messages are.  That message says:
`NameError: name 'not_implemented_for' is not defined`

So, the `not_implmented_for` command is not defined where it is used.
I guess you should use the `nx.utils.not_implemented_for` specification.

Reply:
Unfortunately tests failed again. I clicked on "Details" and can not underastand the error.